env.sample

```diff
@@ -0,0 +1,3 @@
+model__temperature = 0.02
```

Comment:
Keine schlechte Idee. Hatte schon drüber nachgedacht, dass als Schieberegler bei gen in der Frontend einzubauen. Ich werde temperature auch noch mal ausgiebig testen, inwiefern man diesen Wert erhöhen kann, sodass z.B. der Text des Aufenthaltsverlauf 'ausgeschmückt' wird, ohne das kompletter Quatsch generiert wird.

Reply:
Wahrscheinlich ist es dann sinnvoll, die Konfiguration der Modelle von `gen` und `qa` zu trennen. Die Prompts werden sicherlich unterschiedlich sein und evtl. auch max tokens, content_length etc. Damit die einzelnen 'Services' aber nicht wissen müssen wo jetzt genau ihre Parameter verortet werden, würde man die a) via Constructor und b) via Methodenaufruf übergeben. Bzw. kann man auch direkt überlegen inwieweit *Dependency Injection* bzw. *Inversion of Control* sinnvoll sind bei dem Aufbau der Anwendung. Aber ich will jetzt nicht NOCH eine Änderung in den branch packen. Das würde ich dann nach dem Merge angehen.

Reply:
> Hatte schon drüber nachgedacht, dass als Schieberegler bei gen in der Frontend einzubauen. 

Wenn du sowas machen willst, wäre der Workflow a) einen Endpunkt (`set_gen_temperature`) im `Transport` Protokoll anzulegen, dann b) in `Bridge` zu implementieren (was auch nur 'lazy loading' und Weiterleitung an das `gen_backend` bedeuten würde) und schlussendlich c) die eigentliche Übergabe an dein 'service backend'. Klingt viel, ist aber recht straight forward. Mit Python könnte man auch via `__getattr__` ein wenig abkürzen und Methoden dynamisch registrieren, aber das würde das static type checking aushebeln, was wir erst mal nicht machen wollen :).

Reply:
> Konzeptionell wäre die Frage, ob das sinnvoll ist, das in die 'Endanwender'-GUI zu kippen. Die sollte ja möglichst aufgeräumt daherkommen. Eine Idee wäre einen 'developer'/'expert'-modus, wie ihn beispielsweise die Fritz!-Boxen haben. Da muss ein bestimmtes Flag gesetzt sein, damit man bestimmte Eingabemasken sieht.

Ich muss mir dazu noch Gedanken machen, da kommen wir wieder zum Thema UseCase. Ich hatte gedacht, den Regler durch ein paar Tests meinerseits auch zu begrenzen. Es soll nicht die komplette Range von 0.00 bis 1.00 abgedeckt werden, sondern z.B. von 0.05 bis 0.25. Ich glaube es ist keine schlechte Idee dem normalen User Zugang dazu zu verschaffen, weil temperature zu verändern nicht nur anwendungsabhängig Sinn macht, sondern auch dokumentabhängig. Abseits vom Regler ist ja nicht wirklich viel in der Streamlit Oberfläche zu sehen.

Reply:
Ich finde es ganz wichtig, zwischen den Rollen zu unterscheiden: Ist das für *Entwickler:innen* sinnvoll? Ja, absolut! Ist das für *Anwender:innen* sinnvoll? Hängt davon ab. Erwartest du von Nutzenden, dass diese das Konzept von 'Temperatur' verstehen müssen, um die Anwendung nutzen zu können? Für Anwender:innen, wäre es sinnvoll das 'qualitativ' einstellen zu können, also zwischen 'kompakt' und 'ausführlich'. Das lässt sich so aber nicht unbedingt so linear auf Temperatur mappen, sondern hängt evtl. auch vom Prompt ab.